src/components/OnboardingComponents/SemanticSearch/ArticleDialog.tsx

```diff
@@ -0,0 +1,210 @@
+}
+
+/**
+ * A dialog component to display the full content of a selected search result article.
+ */
+export default function ArticleDialog({open, onClose, article}: ArticleDialogProps): JSX.Element {
+  const [numPages, setNumPages] = useState<number>(0);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [containerWidth, setContainerWidth] = useState(0);
+  const [zoom, setZoom] = useState(1);
+  const containerRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const observer = new ResizeObserver((entries: ResizeObserverEntry[]) => {
+      const entry = entries[0];
+      if (entry) {
+        setContainerWidth(entry.contentRect.width);
+      }
+    });
+
+    if (containerRef.current) {
+      observer.observe(containerRef.current);
+    }
+
+    return () => {
+      observer.disconnect();
+    };
+  }, []);
+
+  const goToNextPage = useCallback(() => {
+    setPageNumber((prevPageNumber) => Math.min(prevPageNumber + 1, numPages));
+  }, [numPages]);
+
+  const goToPreviousPage = useCallback(() => {
+    setPageNumber((prevPageNumber) => Math.max(prevPageNumber - 1, 1));
+  }, []);
+
+  const handleZoomIn = useCallback(() => {
+    setZoom((prevZoom) => Math.min(prevZoom + 0.25, 3));
+  }, []);
+
+  const handleZoomOut = useCallback(() => {
+    setZoom((prevZoom) => Math.max(prevZoom - 0.25, 0.25));
+  }, []);
+
+  const handleZoomReset = useCallback(() => {
+    setZoom(1);
+  }, []);
+
+  // Keyboard shortcuts for zoom and navigation
+  useEffect(() => {
+    const handleKeyDown = (event: KeyboardEvent) => {
+      if (!open) return;
+
+      switch (event.key) {
+        case '+':
+        case '=':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            handleZoomIn();
+          }
+          break;
+        case '-':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            handleZoomOut();
+          }
+          break;
+        case '0':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            handleZoomReset();
+          }
+          break;
+        case 'ArrowLeft':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            goToPreviousPage();
+          }
+          break;
+        case 'ArrowRight':
+          if (event.ctrlKey || event.metaKey) {
+            event.preventDefault();
+            goToNextPage();
+          }
+          break;
+      }
+    };
+
+    document.addEventListener('keydown', handleKeyDown);
+    return () => {
+      document.removeEventListener('keydown', handleKeyDown);
+    };
+  }, [open, handleZoomIn, handleZoomOut, handleZoomReset, goToPreviousPage, goToNextPage]);
+
+  if (!article) {
+    return <></>;
+  }
+
+  function onDocumentLoadSuccess({numPages: nextNumPages}: {numPages: number}) {
+    setNumPages(nextNumPages);
+    setPageNumber(1); // Reset to first page on new document load
```

Comment:
Usually we try to avoid inline comments unless e.g. documenting a struct in rare occasions.
```suggestion
    // Reset to first page on new document load
    setPageNumber(1);
```

Reply:
Changed